From acf506c1ae0e701f1d6adf02b1cbcc58a78da6cf Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 3 Aug 2015 13:31:45 +0100
Subject: [PATCH 9/9] Report multiple instances of thermald with thd_log_fatal
 (LP: #1432220)

Rather than aborting via the g_lib fatal log path, we should just
report the error via thd_log_error and exit normally via exit(EXIT_FAILURE).
This avoids the trapping in gmessage which is reported as a signal 5
(SIGTRAP) error.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index e66dad3..2b1befb 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -249,7 +249,7 @@ int main(int argc, char *argv[]) {
 	g_log_set_handler(NULL, G_LOG_LEVEL_MASK, thd_logger, NULL);
 
 	if (check_thermald_running()) {
-		thd_log_fatal(
+		thd_log_error(
 				"An instance of thermald is already running, exiting ...\n");
 		exit(EXIT_FAILURE);
 	}
-- 
2.5.0

